### ISSUE TITLE
Search function returns the second result page on default

### ISSUE THREAD

**Issue by onegen-dev** [recallfuture/furaffinity-api#10]
~~I recently updated this API [to 4.3.1] and noticed, that the search function is not working as expected. I played around the issue to find out as much as I could, so here are my findings.~~

*(wiped node_modules and reinstalled everything before testing; tested on two machines: Windows 10 with Node v16.14.0 and NPM 8.3.1 & Ubuntu Server 20.04 with Node v17.0.1 and NPM 8.4.0)*

~~Common tags or quite popular users returned an entire page of results (48), for example `fox`, `@lower koul`, `ych`, `@lower ariadog` or `mammal`. However, some queries just return 0 results, while they give several on the page itself (number of expected results from FA page in brackets):
`@lower crownedknightdragon` (32), `@lower nikotaru` (13), `@lower larion` (9), `radhim` (8), `sovkac` (6), `trace stark wolf` (7)~~

~~These tags above worked just fine before - there is my account and tags used my friends, so I used them for testing before.~~

**EDIT:** So it seems the Search() functions works correctly, but it for some reason returns the second page on default. This happens when page is not set, but also when it's set to 0 or 1.

This example shows it well:

```javascript
Search("@lower nikotaru", { type: SearchType.Art, rating: Rating.General })
    .then(result => {
        console.log(result.length); // 0
        console.log(result); // [ prev: [Function (anonymous)] ]
        result.prev()
            .then(result2 => {
                console.log(result2.length) // 14
            });
    });
```

So that's pretty much it. If you won't be able to replicate this issue, please say so. If the problem is on my side, I will be thankful for any advice how to solve this problem.

Thanks in advance,
\- OnegenSK

**Comment by recallfuture** [comment]
Yes, page has some problems, and also, I found FA search params has a little changed, that's why it can't get the correct result list.

Thanks for issue, a fixed version is already on the way.

**Comment by recallfuture** [comment]
[main changed params](https://github.com/recallfuture/furaffinity-api/commit/77e97ed1ec46aa670d8f8667f7273606f6743b66#diff-20585e09e483e3122fceaa5ad9b3746de14b6befacd718880ac64e97b46ee7c4R103-R111)

**Comment by recallfuture** [comment]
### 4.3.3

### Fix

- Fix Search [#10](https://github.com/recallfuture/furaffinity-api/issues/10)

### Change

- Remove option "prev" from SearchOptions
- Add full SearchOptions

```js
export interface SearchOptions {
  /** start at 1 */
  page?: number;
  rating?: Rating;
  type?: SearchType;

  // new options
  /** default 'relevancy' */
  orderBy?: OrderBy;
  /** default 'desc' */
  orderDirection?: OrderDirection;
  /** default 'all' */
  range?: RangeType;
  range_from?: Date;
  range_to?: Date;
  /** default extended */
  matchMode?: MatchMode;
}
```
